packages/foam-vscode/src/test/support/vscode-environment.js

```diff
@@ -1,27 +1,27 @@
-// Based on https://github.com/svsool/vscode-memo/blob/master/src/test/env/ExtendedVscodeEnvironment.js
-const VscodeEnvironment = require('jest-environment-vscode');
+const NodeEnvironment = require('jest-environment-node');
 const vscode = require('vscode');
 
-const initialVscode = vscode;
-class ExtendedVscodeEnvironment extends VscodeEnvironment {
+class VscodeEnvironment extends NodeEnvironment {
   async setup() {
     await super.setup();
+    this.global.vscode = vscode;
+
     // Expose RegExp otherwise document.getWordRangeAtPosition won't work as supposed.
     // Implementation of getWordRangeAtPosition uses "instanceof RegExp" which returns false
     // due to Jest running tests in the different vm context.
     // See https://github.com/nodejs/node-v0.x-archive/issues/1277.
     // And also https://github.com/microsoft/vscode-test/issues/37#issuecomment-700167820
     this.global.RegExp = RegExp;
-    this.global.vscode = vscode;
 
     vscode.workspace
       .getConfiguration()
       .update('foam.edit.linkReferenceDefinitions', 'off');
   }
+
   async teardown() {
-    this.global.vscode = initialVscode;
+    this.global.vscode = {};
```

Comment:
The reason for this is that:

```ts
this.global.vscode = initialVscode;
await super.teardown();
```

actually ended up with `this.global.vscode = {}` since that's what the base class did during `teardown`.

Moreover, I fail to understand how can it make a difference, or how can an initial vscode global exist that's different from the current one but still somehow valid.